src/main/java/apoc/neighbors/Neighbors.java

```diff
@@ -0,0 +1,240 @@
+package apoc.neighbors;
+
+import apoc.result.*;
+import org.neo4j.graphdb.*;
+import org.neo4j.helpers.collection.Pair;
+import org.neo4j.procedure.*;
+import org.roaringbitmap.RoaringBitmap;
+
+import java.util.*;
+import java.util.stream.*;
+
+import static apoc.path.RelationshipTypeAndDirections.parse;
+
+public class Neighbors {
+
+    @Context
+    public GraphDatabaseService db;
+
+    @Procedure("apoc.neighbors")
+    @Description("apoc.neighbors(node, rel-direction-pattern, distance) - returns distinct nodes of the given relationships in the pattern up to a certain distance, can use '>' or '<' for all outgoing or incoming relationships")
+    public Stream<NodeResult> neighbors(@Name("node") Node node, @Name(value = "types", defaultValue = "") String types, @Name(value="distance", defaultValue = "1") Long distance) {
+        if (distance < 1) return Stream.empty();
+        if (types==null || types.isEmpty()) return Stream.empty();
+
+        // Initialize bitmaps for iteration
+        RoaringBitmap seen = new RoaringBitmap();
+        RoaringBitmap nextA = new RoaringBitmap();
+        RoaringBitmap nextB = new RoaringBitmap();
+        int nodeId = (int) node.getId();
```

Comment:
I think that's a unsafe operation in case we're working on a graph with more than 2bn nodes - which even is possible when using standard store format. Shouldn't we fall back on either aborting the operation with an exception or somehow deal gracefully with larger node ids?

Reply:
There is NeighborsLarge for those rare folks.

Reply:
Do we want to just drop the regular and rename the .large so it won't be an issue?

Reply:
Maybe leave both as they are. But the int version should throw an execption in case you're using node ids > 2bn. This way we have both, efficency of the int variant and scale of the long variant. just my 2ct